File: Devices.Web.API/Models/Entities/Device.cs
using System;
using System.Collections.Generic;

namespace Devices.Web.API.Models.Entities
{
    public partial class Device
    {
        public Device()
        {
            DeviceUsages = new HashSet<DeviceUsage>();
        }

        public int DeviceId { get; set; }
        public int DeviceTypeId { get; set; }
        public int DeviceStatusId { get; set; }
        public string Name { get; set; } = null!;
        public short Temperature { get; set; }

        public virtual DeviceStatus DeviceStatus { get; set; } = null!;
        public virtual DeviceType DeviceType { get; set; } = null!;
        public virtual ICollection<DeviceUsage> DeviceUsages { get; set; }
    }
}

File: Devices.Web.UI/ClientApp/src/app/view_models/device.ts

export class Device {
    deviceId!: number;
    deviceTypeId!: number;
    deviceStatusId!: number;
    name!: string;
    temperature!: number;
    usage?: number [];
    deviceTypeName!: string;
    deviceStatusName!: string;

    constructor(_deviceId: number, _deviceTypeId: number, _deviceStatusId: number, _name: string, _temperature: number, _deviceTypeName: string, _deviceStatusName: string, _usage?: number[]) {
        this.deviceId = _deviceId;
        this.deviceTypeId = _deviceTypeId;
        this.deviceStatusId = _deviceStatusId;
        this.name = _name;
        this.temperature = _temperature;
        this.usage = this.usage;
        this.deviceTypeName = _deviceTypeName;
        this.deviceStatusName = _deviceStatusName;
     }
}